pkg/eks/filecache.go

```diff
@@ -0,0 +1,228 @@
+package eks
```

Comment:
why is this in the `eks` package? does it call the eks api?

Reply:
Why wouldn't it be? It's the other way around. eks is using this thing. I mean it can live in a separate package, sure if you'd prefer.

Reply:
doesn't bother me really if there is a logical reason for it to be there. we just tend to overload that package a bit and oftentimes i end up there like "how did i even get here?"

Reply:
:D That's a reasonable thought to me! :) To understand why I put it there.. It's dealing with the cred coming back from the API, right? So I thought it's reasonable to put it next to it.

But I also thought of just putting it into a cache package, or creds package, or providers package... What do you think would be a nice package name for it then? apicache, credcache, provider, cache...?

Reply:
> It's dealing with the cred coming back from the API, right? So I thought it's reasonable to put it next to it.

that makes sense.

and yeh the trouble of moving to its own package is that we are leaning towards a two-word name 😂 . I like `credcache` best simply because it is most descriptive 🤷‍♀️  or maybe just `credential`

Reply:
Done :)